src/barril/units/_quantity.py

```diff
@@ -660,9 +660,17 @@ def Convert(self, value, to_unit):
         :returns:
             An object with values to the passed unit.
         """
-        return self._unit_database.Convert(
-            self._composing_categories, self._composing_units, to_unit, value
-        )
+        try:
+            return self._unit_database.Convert(
+                self._composing_categories, self._composing_units, to_unit, value
+            )
+        except:
+            return self._unit_database.Convert(
+                self._category,
+                self._CreateUnitsWithJoinedExponentsString(),
+                to_unit,
+                value,
+            )
```

Comment:
In case that anyone did not notice, this is clearly a hackish solution. The original `Convert` passing `_composing_categories` and `_composing_units` does not work after modifying `__hash__` and `__eq__` implementations. However if we use `self._category` and `self._CreateUnitsWithJoinedExponentsString()` it works on everything BUT one test (which is a clearly a strange one for me):

```python
def testEmptyScalar():
    """
        ScalarMultiData exception when some of its scalars don't have a quantity_type
    """
    # An empty scalar doesn't have a category defined
    scalar_1 = Scalar.CreateEmptyScalar(20.0)

    # When try to retrieve scalar value or unit a  exception was being raised
    assert scalar_1.GetValue("m") == 20.0
    assert scalar_1.GetUnit() == ""
```

A few comments about this test:

- I don't think `scalar_1.GetValue("m")` should work since `scalar_1` is an `EmptyScalar`, not sure which project relies on this behavior.
- In the docstring, don't know what `ScalarMultiData` exception is, couldn't find anywhere

If we use:

```python
return self._unit_database.Convert(
    self._category,
    self._CreateUnitsWithJoinedExponentsString(),
    to_unit,
    value,
)
```

then that test crashes.

Reply:
I support we change the behavior, this API is clearly unexpected